ggml/src/ggml-cuda/cpy.cu

```diff
@@ -310,6 +514,8 @@ void ggml_cuda_cpy(ggml_backend_cuda_context & ctx, const ggml_tensor * src0, gg
     char * src1_ddc = (char *) src1->data;
 
     const bool contiguous_srcs = ggml_is_contiguous(src0) && ggml_is_contiguous(src1);
+    const bool can_be_transposed = src0->op == GGML_OP_TRANSPOSE && !ggml_is_contiguous(src0) &&
+        (src0->ne[3] == 1 || (src0->nb[2] <= src0->nb[3] && src0->nb[0] < src0->nb[2]));
```

Comment:
Shouldn't this be something like this?

```suggestion
    const bool can_be_transposed = src0->nb[1] == ggml_element_size(src0);
```

As long as the buffer is contiguous in dimension 1 you should always be able to load it in a transposed way.